src/EditorFeatures/CSharpTest/GenerateDefaultConstructors/GenerateConstructorForAbstractClass.cs

```diff
@@ -0,0 +1,318 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.CodeRefactorings;
+using Microsoft.CodeAnalysis.GenerateDefaultConstructors;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.GenerateDefaultConstructors
+{
+    public class GenerateConstructorForAbstractClassTests : AbstractCSharpCodeActionTest
+    {
+        protected override CodeRefactoringProvider CreateCodeRefactoringProvider(Workspace workspace, TestParameters parameters)
+            => new GenerateDefaultConstructorsCodeRefactoringProvider();
+
+        [WorkItem(25238, "https://github.com/dotnet/roslyn/issues/25238")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateDefaultConstructors)]
+        public async Task TestGenerateConstructorFromProtectedConstructor()
+        {
+            await TestInRegularAndScriptAsync(
+@"abstract class C : [||]B
```

Comment:
you don't have to change up all these test.  But i found them odd to read as you had to read out of order.  isntead of seeing hte base type, then the dependent type, you see the dependent type and then have to find the base type.

Reply:
Yeah, I agree. The tests were written in the same way as in GenerateDefaultConstructorsTests.cs.